proposals/NNNN-Promises-in-Policy-Server.md

```diff
@@ -0,0 +1,114 @@
+# Use Promises in Server Logic of SDL Policy Server
+
+* Proposal: [SDL-NNNN](NNNN-filename.md)
+* Author: [Chris Rokita](https://github.com/crokita)
+* Status: **Awaiting review**
+* Impacted Platforms: Policy Server
+
+## Introduction
+
+This proposal aims to change the majority of the server application logic from using callback-based to Promise-based asynchronous function management. This is an effort to modernize the codebase after years of stagnant structural changes to make the project more readable and easier to update when new features are requested.
+
+## Motivation
+
+The last structural change to the policy server was written at a time where Promises were not as well understood, and so the established method of handling asynchronous tasks was using callbacks. Earlier attempts to mitigate the issues of readability with this method have not scaled so well with the number of features the project has accumulated. The codebase has grown to a point where any additions or changes to the application logic require a significant amount of reading and care to ensure that no new bugs are introduced and that all cases are considered.
+
+With the introduction of await/async syntax, asynchronous logic can be read from top to bottom without excessive nesting of functions or visually jumping around the method to keep track of the execution order. This Promise style syntax will require fewer lines and will automatically throw functions into the event loop instead of being ran synchronously.
```

Comment:
```suggestion
With the introduction of await/async syntax, asynchronous logic can be read from top to bottom without excessive nesting of functions or visually jumping around the method to keep track of the execution order. This Promise style syntax will require fewer lines and will automatically throw functions into the event loop instead of being run synchronously.
```